Spring-server/src/main/java/ikpi63holding/postclient/controllers/FileController.java

```diff
@@ -23,22 +26,26 @@
     @Autowired
     private ServletContext servletContext;
 
-    private final MailboxRepository repository;
+    private final UserRepository userRepository;
 
-    FileController(MailboxRepository repository) {
-        this.repository = repository;
+    FileController(UserRepository userRepository) {
+        this.userRepository = userRepository;
     }
 
-    @GetMapping("/files/{id}/{folder}/{messageId}/{filename}")
+    @GetMapping("/files/{userName}/{mailboxId}/{folder}/{messageId}/{filename}")
     @ResponseBody
-    public ResponseEntity<InputStreamResource> serveFile(@PathVariable long id,
+    public ResponseEntity<InputStreamResource> serveFile(@PathVariable String userName,
+            @PathVariable int mailboxId,
             @PathVariable String folder, @PathVariable int messageId, @PathVariable String filename)
             throws Exception {
         MediaType mediaType = MediaTypeUtils.getMediaTypeForFileName(this.servletContext, filename);
-        Mailbox mailbox = repository.findById(id).orElse(null);
+        User user = userRepository.findByUsername(userName).orElseThrow(
+                () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "No such user"));
+
+        Mailbox mailbox = user.getMailboxes().get(mailboxId - 1);
```

Comment:
magic number

Reply:
Ok, this I missed, but it is to be reworked next.